early-boot-gui-2020-02-05-aarch64-unknown-linux-gnu.drv' failed with exit code 2

When building `nix-build examples/hello --argstr device oneplus-oneplus3 -A build.default`

on:
```
commit 322fdb9847e34b4cf40dfabfba5d470ca163d592 (HEAD -> master, origin/master, origin/HEAD)
Merge: e689208 6e00fa0
Author: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date:   Sun Jun 28 23:53:07 2020 -0400

    Merge pull request #171 from samueldr-wip/device/xiaomi-begonia

    device: add **xiaomi-begonia**
```
and
```
NixOS 20.09pre231796.22a81aa5fc1
```

I get:

```
  CC introspection.c
  CC hal.c
  lvgui_struct_accessors.c:4:14: warning: no previous prototype for 'lvgui_allocate_lv_style' [-Wmissing-prototypes]
      4 | lv_style_t * lvgui_allocate_lv_style() {
        |              ^~~~~~~~~~~~~~~~~~~~~~~
  CC artwork/lvgui_cursor.c
  CC lvgui_struct_accessors.c
  CC /build/source/lvgl/src/lv_draw/lv_draw_rect.c
  CC lv_lib_nanosvg/lv_nanosvg.c
  builder for '/nix/store/xyk090lik3p60nhaaq43gpp5aww9ajmr-mobile-nixos-early-boot-gui-2020-02-05-aarch64-unknown-linux-gnu.drv' failed with exit code 2
```

--- a/overlay/mruby-builder/mrbgems/mruby-lvgui/lvgui.nix
+++ b/overlay/mruby-builder/mrbgems/mruby-lvgui/lvgui.nix
@@ -24,14 +24,14 @@ let
 in
   stdenv.mkDerivation {
     pname = "mobile-nixos-early-boot-gui";
-    version = "2020-02-05";
+    version = "2020-07-02";
 
     src = fetchFromGitHub {
       fetchSubmodules = true;
       repo = "lvgui";
       owner = "mobile-nixos";
-      rev = "d98d5e59ba0f4a76b2f092ee957a198d9e749dfb";
-      sha256 = "1hn01mi44wmx12987va4h69ldnvjbvniwm9slnd5naib6j2n5rbw";
+      rev = "d531b45b19612e520eec26660029c49f223d8d3a";
+      sha256 = "0cyra0ji2sb48c9h2vafzzsfz3y26b6k37mfc6gnv0lfx0js52hs";
     };
 
     # Document `LVGL_ENV_SIMULATOR` in the built headers.

--- a/overlay/mruby-builder/mruby/default.nix
+++ b/overlay/mruby-builder/mruby/default.nix
@@ -18,6 +18,7 @@ in
 , file
 , mruby
 , writeText
+, writeShellScriptBin
 
 # When unset the default gembox will be used.
 # For a native build, use the default gembox
@@ -145,6 +146,16 @@ let
     end
     ''}
   '';
+
+  # Inspired from #91991
+  # https://github.com/NixOS/nixpkgs/pull/91991
+  # The mruby build would need to be patched in the future.
+  # As 2.2 will require other invasive changes, this is worked around until 2.2 is released.
+  # The default (without other inputs) mruby build does not use `pkg-config`,
+  # but its `#search_package` implementation does.
+  pkgconfig-helper = writeShellScriptBin "pkg-config" ''
+    exec ${buildPackages.pkgconfig}/bin/${buildPackages.pkgconfig.targetPrefix}pkg-config "$@"
+  '';
 in
 stdenv.mkDerivation rec {
   pname = "mruby";
@@ -163,7 +174,7 @@ stdenv.mkDerivation rec {
     ./bison-36-compat.patch
   ];
 
-  nativeBuildInputs = [ ruby bison rake ] ++ gemNativeBuildInputs;
+  nativeBuildInputs = [ pkgconfig-helper ruby bison rake ] ++ gemNativeBuildInputs;
   buildInputs = gemBuildInputs;
 
   # Necessary so it uses `gcc` instead of `ld` for linking.

